triggers/runner/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerRunnerConfig.scala

```diff
@@ -14,45 +36,48 @@ import scala.concurrent.duration._
   * @param inFlightCommandBackPressureCount When the number of in-flight command submissions exceeds this value, then we
   *                                         enable Daml rule evaluation to apply backpressure (by failing emitCommands
   *                                         evaluations).
-  * @param inFlightCommandOverflowCount When the number of in-flight command submissions exceeds this value, then we
-  *                                     kill the trigger instance by throwing an InFlightCommandOverflowException.
-  * @param allowInFlightCommandOverflows flag to control whether we allow in-flight command overflows or not.
   * @param submissionFailureQueueSize Size of the queue holding ledger API command submission failures.
   * @param maximumBatchSize Maximum number of messages triggers will batch (for rule evaluation/processing).
   * @param batchingDuration Period of time we will wait before emitting a message batch (for rule evaluation/processing).
-  * @param maximumActiveContracts Maximum number of active contracts that we will store at any point in time.
   */
 final case class TriggerRunnerConfig(
     parallelism: Int,
     maxRetries: Int,
     maxSubmissionRequests: Int,
     maxSubmissionDuration: FiniteDuration,
     inFlightCommandBackPressureCount: Long,
-    inFlightCommandOverflowCount: Int,
-    allowInFlightCommandOverflows: Boolean,
     submissionFailureQueueSize: Int,
     maximumBatchSize: Long,
     batchingDuration: FiniteDuration,
-    maximumActiveContracts: Long,
+    hardLimit: TriggerRunnerHardLimits,
 )
 
 object TriggerRunnerConfig {
   val DefaultTriggerRunnerConfig: TriggerRunnerConfig = {
     val parallelism = 8
+    val maxSubmissionRequests = 100
+    val maxSubmissionDuration = 5.seconds
 
     TriggerRunnerConfig(
       parallelism = parallelism,
       maxRetries = 6,
-      maxSubmissionRequests = 100,
-      maxSubmissionDuration = 5.seconds,
+      maxSubmissionRequests = maxSubmissionRequests,
+      maxSubmissionDuration = maxSubmissionDuration,
       inFlightCommandBackPressureCount = 1000,
-      inFlightCommandOverflowCount = 10000,
-      allowInFlightCommandOverflows = true,
       // 256 here comes from the default ExecutionContext.
       submissionFailureQueueSize = 256 + parallelism,
       maximumBatchSize = 1000,
       batchingDuration = 250.milliseconds,
-      maximumActiveContracts = 10000,
+      hardLimit = TriggerRunnerHardLimits(
+        maximumActiveContracts = 10000,
+        inFlightCommandOverflowCount = 10000,
+        allowInFlightCommandOverflows = true,
+        // 50% extra on the maxSubmissionDuration value
+        ruleEvaluationTimeout = maxSubmissionDuration * 3 / 2,
+        // 50% extra on mean time between submission requests
+        stepInterpreterTimeout = (maxSubmissionDuration / maxSubmissionRequests.toLong) * 3 / 2,
+        allowTriggerTimeouts = false,
```

Comment:
By default, we disable trigger timeouts since we have yet to understand how they impact existing code

Reply:
Can we not use a very long timeout (e.g. `Long.MaxValue`) instead ?

Reply:
Absolutely, this could be done. However, I felt it would be more honest (in this instance) to have an explicit flag for disabling/enabling these timeouts.